### PR TITLE
helm chart: Add Pod Security Policy configs

### PR DESCRIPTION
**Pull request by surajssd**
This commit adds PodSecurityPolicy, Role and RoleBinding configs. And a
flags in values file which decides whether the PSP and related configs
will be installed or not.

Signed-off-by: Suraj Deshmukh <surajd.service@gmail.com>

### CODE REVIEW

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{{- if .Values.podSecurityPolicy }}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: mysql-operator
```
Comment (calind) [inline hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]:
```suggestion
  name: '{{ template "mysql-operator.fullname" . }}'
  labels:
    app: '{{ template "mysql-operator.name" . }}'
    chart: '{{ template "mysql-operator.chart" . }}'
    heritage: '{{ .Release.Service }}'
    release: '{{ .Release.Name }}'
```

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{{- if .Values.podSecurityPolicy }}
```
Comment (calind) [inline hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]:
```suggestion
{{- if .Values.podSecurityPolicy.enabled }}
```

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: mysql-operator
+  annotations:
```
Comment (calind) [inline hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]: I think annotations should be changable. Maybe define a `.Values.podSecurityPolicy.annotations` like some charts in https://github.com/helm/charts do.

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  - SETGID
+  # Allow core volume types.
+  volumes:
+  - 'configMap'
+  - 'emptyDir'
+  - 'projected'
+  - 'secret'
+  - 'downwardAPI'
+  # Assume that persistentVolumes set up by the cluster admin are safe to use.
+  - 'persistentVolumeClaim'
```
Comment (calind) [inline hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]: I don't tink we need pvcs for mysql-operator.

File: hack/charts/mysql-operator/templates/role.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.podSecurityPolicy }}
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mysql-operator
```
Comment (calind) [inline hack/charts/mysql-operator/templates/role.yaml]:
```suggestion
  name: '{{ template "mysql-operator.fullname" . }}'
  labels:
    app: '{{ template "mysql-operator.name" . }}'
    chart: '{{ template "mysql-operator.chart" . }}'
    heritage: '{{ .Release.Service }}'
    release: '{{ .Release.Name }}'
```

File: hack/charts/mysql-operator/templates/role.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+rules:
+- apiGroups: ['policy']
+  resources: ['podsecuritypolicies']
+  verbs:     ['use']
+  resourceNames:
+  - mysql-operator
```
Comment (calind) [inline hack/charts/mysql-operator/templates/role.yaml]:
```suggestion
  - '{{ template "mysql-operator.fullname" . }}'
```

File: hack/charts/mysql-operator/templates/rolebinding.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.podSecurityPolicy }}
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mysql-operator
```
Comment (calind) [inline hack/charts/mysql-operator/templates/rolebinding.yaml]:
```suggestion
  name: '{{ template "mysql-operator.fullname" . }}'
  labels:
    app: '{{ template "mysql-operator.name" . }}'
    chart: '{{ template "mysql-operator.chart" . }}'
    heritage: '{{ .Release.Service }}'
    release: '{{ .Release.Name }}'
```

File: hack/charts/mysql-operator/templates/rolebinding.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+roleRef:
+  kind: Role
+  name: mysql-operator
```
Comment (calind) [inline hack/charts/mysql-operator/templates/rolebinding.yaml]:
```suggestion
  name: '{{ template "mysql-operator.fullname" . }}'
```

File: hack/charts/mysql-operator/values.yaml [hunk 9]
```diff
@@ -26,6 +26,8 @@ affinity: {}
 
 extraArgs: []
 
+podSecurityPolicy: false
```
Comment (calind) [inline hack/charts/mysql-operator/values.yaml]:
```suggestion
podSecurityPolicy:
  enabled: false
  annotations:
    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
```

File: hack/charts/mysql-operator/templates/role.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.podSecurityPolicy }}
```
Comment (calind) [inline hack/charts/mysql-operator/templates/role.yaml]:
```suggestion
{{- if and .Values.podSecurityPolicy.enabled .Values.rbac.create }}
```

File: hack/charts/mysql-operator/templates/rolebinding.yaml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.podSecurityPolicy }}
```
Comment (calind) [inline hack/charts/mysql-operator/templates/rolebinding.yaml]:
```suggestion
{{- if and .Values.podSecurityPolicy.enabled .Values.rbac.create }}
```

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml (hunk 4 above)
Comment (surajssd) [reply hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]: removed

File: hack/charts/mysql-operator/templates/podsecuritypolicy.yaml (hunk 3 above)
Comment (surajssd) [reply hack/charts/mysql-operator/templates/podsecuritypolicy.yaml]: done

**Pull request merged**